aws_requests_auth/boto_utils.py

```diff
@@ -31,7 +31,7 @@ def get_credentials(credentials_obj=None):
 
 class BotoAWSRequestsAuth(AWSRequestsAuth):
 
-    def __init__(self, aws_host, aws_region, aws_service):
+    def __init__(self, aws_host, aws_region, aws_service, session_params={}):
```

Comment:
thanks for the pull request @dashstander . Mutable default kwargs are a bit of a Python gotcha e.g.

```python
In [1]: def f(l=[]): 
   ...:     l.append(1) 
   ...:     print(l) 
   ...:                                                                                                                               

In [2]: f()                                                                                                                           
[1]

In [3]: f()                                                                                                                           
[1, 1]

In [4]: f()                                                                                                                           
[1, 1, 1]

In [5]: f()                                                                                                                           
[1, 1, 1, 1]

In [6]: f()                                                                                                                           
[1, 1, 1, 1, 1]
```

The preferred pattern ([as described by core dev Raymond Hettinger](https://twitter.com/raymondh/status/822296239156920321)) looks something like...

```python
def f(l=None):
    if l is None:
        l = []
```

---

> allows things like passing in regions, profiles, etc.

Since `aws_region` is "hardcoded" in `BotoAwsRequestsAuth` and `AWSRequestsAuth`, I do not think `aws_region` will be an appropriate target for us to dynamically override. 

I'm not sure if this is true, but does the `boto3` `profile_name` kwarg (e.g. `boto3.session.Session(profile_name="my-profile")`) restrict itself to only  affecting`aws_access_key_id`, `aws_secret_access_key`, `aws_session_token`? 

If `profile_name` indeed only can possibly affect auth, then we might consider allowing `profile_name` (and only `profile_name`) to be optionally provided instead.

Reply:
I appreciate the response @DavidMuller , this is the first time I've made a PR against a public repo.

Having the default be mutable was definitely an oversight on my part and I can change the PR.

According to the [docs I'm looking at](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-profiles.html),  there are other parameters (including `aws_region`) that can have defaults configured by different profiles. But I don't think that that would interfere with having it `aws_region` hardcoded. The `Session().get_credentials()` call won't ever pull anything except the `aws_access_key_id`, `aws_secret_access_key`, `aws_session_token`, so there's no danger of the `aws_region` getting overridden or anything like that.

Which is all another way of saying that adding an optional `profile_name` parameter works for me as well.